spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -105,8 +105,49 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] {
           return withInfos(scanExec, fallbackReasons.toSet)
         }
 
-        val scanImpl = COMET_NATIVE_SCAN_IMPL.get()
-        if (scanImpl == CometConf.SCAN_NATIVE_DATAFUSION && !COMET_EXEC_ENABLED.get()) {
+        var scanImpl = COMET_NATIVE_SCAN_IMPL.get()
```

Comment:
Should we also log a message to indicate that auto scan  was enabled (and the scan implementation it selected)?

Reply:
Yes, that is a good idea. I will add that today.